File: C++/Step15-Stack/10828.cpp
//
//  10828.cpp
//  Step15-Stack
//
//  Created by 김예빈 on 2019. 9. 18..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
#include <string>
using namespace std;

int stack[10000];
int top = -1;

void push(int);
int pop();
bool empty();

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, num;
    string ir;
    
    cin >> n;
    
    while(n--) {
        cin >> ir;
        
        if(ir == "push") {
            cin >> num;
            push(num);
        }
        
        if(ir == "pop")
            cout << pop() << "\n";
        
        if(ir == "size")
            cout << top + 1 << "\n";
        
        if(ir == "empty")
            cout << empty() << "\n";
        
        if(ir == "top") {
            if(empty())
                cout << "-1\n";
            else
                cout << stack[top] << "\n";
        }
    }
    
    return 0;
}

void push(int input) {
    stack[++top] = input;
}

int pop() {
    int num = stack[top];
    
    if (top < 0)
        return -1;
    else {
        stack[top--] = 0;
        return num;
    }
}

bool empty() {
    if(top < 0)
        return 1;
    else
        return 0;
}

File: BaekJoon Online Judge/Swift/Step17-Stacks/10828.swift
//
//  10828.swift
//  Step17-Stacks
//
//  Created by Yebin Kim on 2019/12/23.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

public struct Stack<T> {
    private var stack = Array<T>()
    
    public mutating func push(_ element: T) {
        stack.append(element)
    }
    
    public mutating func pop() -> T {
        if stack.isEmpty {
            return -1 as! T
        } else {
            return stack.popLast()!
        }
    }
    
    public var size: Int {
        return stack.count
    }
    
    public var empty: Int {
        if stack.isEmpty {
            return 1
        } else {
            return 0
        }
    }
    
    public var top: T {
        if stack.isEmpty {
            return -1 as! T
        } else {
            return stack.last!
        }
    }
}

var stack = Stack<Int>()

for _ in 0..<Int(readLine()!)! {
    let op = readLine()!.split(separator: " ")
    
    switch op[0] {
    case "push":
        stack.push(Int(String(op[1]))!)
        break
        
    case "pop":
        print(stack.pop())
        break
        
    case "size":
        print(stack.size)
        break
        
    case "empty":
        print(stack.empty)
        break
        
    case "top":
        print(stack.top)
        break
        
    default: break
    }
}
